spyder/plugins/editor/lsp/manager.py

```diff
@@ -82,7 +82,7 @@ def get_language_config(self, language):
         """Get language configuration options from our config system."""
         if language == 'python':
             self.update_python_config()
-            return self.python_config
+            return self.python_config.copy()
```

Comment:
Why is this needed? I mean, it shouldn't be necessary.

Reply:
While testing the changes, the restart was not being triggered (the `current_config` and `new_config` in `update_server_list` were the same). This was caused because the client for `python` was being affected for the change in `self.python_config` (adding the `copy` prevents this to happen)

Reply:
Ok, I understand what you mean now and that's a serious problem because by using a shared `self.python_config` attribute we are modifying the configuration of the existing Python client and of future clients too.

To avoid that, please do the following:

1. Remove `self.python_config` from `LSPManager`
2. Rename `update_python_config` to `generate_python_config`
3. Use a local variable inside that method called `python_config`, like this:

    ```python
    def generate_python_config(self):
    """
    Generate Python server configuration with the options saved in our
    config system
    """
    python_config = PYTHON_CONFIG.copy()
    ...
    ...
    return python_config
    ```

    and use `python_config` instead of `self.python_config` in that method.